### PR TITLE
Correctly respect namespace when routing and filtering group members

### PR DESCRIPTION
**Pull request by kuangp**
Signed-off-by: Phil Kuang <pkuang@factset.com>

## Hey, I just made a Pull Request!

Correct the routing to group members when they exist in a different namespace from the group entity.
Correctly filter group members while respecting the target namespace in the relation.
Previously if you were a member of a group in a certain namespace, every duplicate group in other namespaces will include you in their membership list.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/org/src/components/Cards/Group/MembersList/MembersListCard.tsx [hunk 1]
```diff
@@ -117,13 +111,15 @@ export const MembersListCard = (_props: {
...
     spec: { profile },
   } = groupEntity;
   const catalogApi = useApi(catalogApiRef);
 
   const displayName = profile?.displayName ?? groupName;
 
+  const groupNamespace = grpNamespace || ENTITY_DEFAULT_NAMESPACE;
+
   const { loading, error, value: members } = useAsync(async () => {
     const membersList = await catalogApi.getEntities({
```
Comment (dhenneke) [inline plugins/org/src/components/Cards/Group/MembersList/MembersListCard.tsx:124]: This whole filtering logic could be replaced by [`useRelatedEntities` from `plugin-catalog-react`](https://github.com/backstage/backstage/blob/c36a7847e49a1b61f76cff4451d30461c96e1246/plugins/catalog-react/src/hooks/useRelatedEntities.ts):

```ts
const groupMembersList = useRelatedEntities(groupEntity, {type: RELATION_MEMBER_OF, kind: 'User'});
```

But I'm not sure if the implementation is more efficient (getting all `User`s and filter them vs. getting all relations and query all users individually). I guess it depends on how many users are in an average sized group. @freben thoughts?

Comment (freben) [reply plugins/org/src/components/Cards/Group/MembersList/MembersListCard.tsx:124]: We really need https://github.com/backstage/backstage/issues/4499 ...

Yeah use this code for now. I see the hook doesn't limit the issue concurrency, and before adopting it more maybe we should make the code issue batch queries via the getEntities method instead. (Giving multiple values to `filter`, one for each kind:namespace/name triplet).

Comment (freben) [reply plugins/org/src/components/Cards/Group/MembersList/MembersListCard.tsx:124]: Another thing we could do to lighten the load is to use the `fields` parameter to the api, to only return the exact entity fields that we need to render the grid items.

**Review by freben — APPROVED** [review]

File: plugins/org/src/components/Cards/Group/MembersList/MembersListCard.tsx (hunk 1 above)
Comment (freben) [reply plugins/org/src/components/Cards/Group/MembersList/MembersListCard.tsx:124]: But this is good to go as far as I can tell!

**Pull request merged**